lib/cocoapods/installer.rb

```diff
@@ -456,7 +456,22 @@ def clean_sandbox(pod_targets)
         title_options = { :verbose_prefix => '-> '.red }
         sandbox_state.deleted.each do |pod_name|
           UI.titled_section("Removing #{pod_name}".red, title_options) do
-            sandbox.clean_pod(pod_name)
+            root_name = Specification.root_name(pod_name)
+            pod_dir = sandbox.local?(root_name) ? nil : sandbox.pod_dir(root_name)
+            sandbox.clean_pod(pod_name, pod_dir)
+          end
+        end
+      end
+
+      # Check any changed pods that became local pods and used to be remote pods and
+      # ensure the sandbox is cleaned up.
+      unless sandbox_state.changed.empty?
+        title_options = { :verbose_prefix => '-> '.red }
+        sandbox_state.changed.each do |pod_name|
+          UI.titled_section("Cleaning #{pod_name}".red, title_options) do
+            previous_spec_repo = sandbox.manifest.spec_repo(pod_name)
```

Comment:
should we also be checking for other checkout options such as `git:` or does this only occur when going from spec repo -> local?

Reply:
form spec repo -> local, I think switching sources works correctly.

Reply:
The latter works because I think we decide to reinstall the pod which eventually cleans it up.